cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowContext.java

```diff
@@ -35,16 +47,46 @@
   private final ProgramWorkflowRunner programWorkflowRunner;
   private final Map<String, String> runtimeArgs;
   private final WorkflowToken token;
+  private final Metrics userMetrics;
 
   BasicWorkflowContext(WorkflowSpecification workflowSpec, @Nullable WorkflowActionSpecification specification,
                        long logicalStartTime, @Nullable ProgramWorkflowRunner programWorkflowRunner,
                        Map<String, String> runtimeArgs, WorkflowToken token) {
+    this(workflowSpec, specification, logicalStartTime, programWorkflowRunner,
+         ImmutableMap.copyOf(runtimeArgs), token, null, null, null, null, null, null);
+  }
```

Comment:
In `WorkflowDriver` when we execute the condition, we create the `BasicWorkflowContext` and pass this context to the predicate inside the condition. Since context now has `getDataset` method, I think its better to wrap the `predicate.apply(context)` inside transaction as well, so that conditions will also have access to the datasets and services.